custom/icds_reports/utils/aggregation_helpers/distributed/location_reassignment.py

```diff
@@ -0,0 +1,163 @@
+    CREATE UNLOGGED TABLE "{prev_local}" AS (SELECT * FROM "{current_table}" WHERE location_reassignment_date='{prev_month}');
+    UPDATE "{prev_local}" SET supervisor_id = last_supervisor_id, awc_id=last_owner_id;
+    DELETE FROM "{prev_table}" WHERE location_reassignment_date='{prev_month}';
+    INSERT INTO "{prev_table}" (SELECT * FROM "{prev_local}");
+    """
+
+    def drop_temp_tables(self, alias):
+        data = {
+            'prev_table': get_prev_table(alias),
+            'prev_local': f"{alias}_prev_local",
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.DROP_QUERY.format(**data))
+
+    def create_temp_tables(self, table, day):
+        alias, table = table
+        data = {
+            'prev_table': get_prev_table(alias),
+            'prev_local': f"{alias}_prev_local",
+            'prev_month': day,
+            'current_table': table,
+            'alias': alias
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.CREATE_QUERY.format(**data))
+
+    def make_all_tables(self, day):
+        day = transform_day_to_month(day) + relativedelta(months=1)
+        table_list = [
+            ('static-child_health_cases', get_table_name(DASHBOARD_DOMAIN, 'static-child_health_cases')),
+            ('static-ccs_record_cases', get_table_name(DASHBOARD_DOMAIN, 'static-ccs_record_cases')),
+            ('static-person_cases_v3', get_table_name(DASHBOARD_DOMAIN, 'static-person_cases_v3')),
+            ('static-household_cases', get_table_name(DASHBOARD_DOMAIN, 'static-household_cases')),
+            ('static-child_tasks_cases', get_table_name(DASHBOARD_DOMAIN, 'static-child_tasks_cases')),
+            ('static-pregnant-tasks_cases', get_table_name(DASHBOARD_DOMAIN, 'static-pregnant-tasks_cases')),
+        ]
+        for table in table_list:
+            self.drop_temp_tables(table[0])
+            self.create_temp_tables(table, day)
+
+
+class TempPrevIntermediateTables(TempPrevTables):
+    CREATE_QUERY = """
+    CREATE UNLOGGED TABLE "{prev_table}" (LIKE "{current_table}");
+    SELECT create_distributed_table('{prev_table}', 'supervisor_id');
+    INSERT INTO "{prev_table}" (SELECT * FROM "{current_table}" where month='{prev_month}');
+    CREATE INDEX "idx_sup_case_{alias}" ON "{prev_table}" USING hash (case_id);
+    CREATE INDEX "idx_sup_state_{alias}" ON "{prev_table}" USING hash (state_id);
+    CREATE UNLOGGED TABLE "{prev_local}" AS (SELECT * FROM "{current_table}" WHERE case_id in (select case_id from "{ucr_local}"));
+    DELETE FROM "{prev_table}" WHERE case_id in (select doc_id from "{ucr_local}");
+    UPDATE "{prev_local}" prev SET supervisor_id = last_supervisor_id FROM "{ucr_local}" ucr WHERE prev.case_id=ucr.doc_id;
+    INSERT INTO "{prev_table}" (SELECT * FROM "{prev_local}");
```

Comment:
Are these four lines this bit? Having a hard time wrapping my head around them.

> Creating those tables was a bit trickier than expected since citus does not let you update the shard column, which is why I used DELETE followed by INSERT and did the updates locally on the master node.

Reply:
Yeah, there is a similar version in all three of these classes, but this version is, in order:
1. creates a local copy of all the rows in the previous month's table that need to be updated (were migrated at the start of the next month).
2. delete those same rows from the previous month's table
3. modifies the local copy created in step 1 to update location info to previous owners
4. inserts the local copy data into the previous month table